### PR TITLE
Skeleton of console and new package structure.

### PR DESCRIPTION
**Pull request by afshin**
This PR has the base files for the new console in a `console` directory and it also has two `examples` instead of just one, so it has a slightly updated folder structure. The console does not work, but it's best to have the base files in `master` before straying too far.

### CODE REVIEW

File: examples/console/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+let SERVER_URL = getBaseUrl();
+let NOTEBOOK = 'test.ipynb';
+
+
+function main(): void {
+  // Initialize the keymap manager with the bindings.
+  var keymap = new KeymapManager();
+
+  // Setup the keydown listener for the document.
+  document.addEventListener('keydown', event => {
```
Comment (blink1073) [inline examples/console/src/index.ts:62]: You'll need to pass `true` here to capture the events before CodeMirror does.

Comment (afshin) [inline examples/console/src/index.ts:62]: Thanks!

**Comment by blink1073** [comment]
Nice!

**Comment by ellisonbg** [comment]
Awesome! While the console would reuse parts of the notebook, I would
imagine it being a separate top level plugin. Would it make sense to have a
separate jupyter-js-console package/repo?

On Wed, Apr 13, 2016 at 10:42 AM, Steven Silvester <notifications@github.com

> wrote:
> 
> Merged #175 https://github.com/jupyter/jupyter-js-notebook/pull/175.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/jupyter/jupyter-js-notebook/pull/175#event-626154307

## 

Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by blink1073** [comment]
I would actually prefer there to be only three repos, to mirror what we're doing with Phosphor: `jupyter-js-core`, `jupyter-js-ui`, and `jupyter-lab`, but that is perhaps a conversation for Friday.

**Comment by afshin** [comment]
Since it's going to use so many notebook things, I think for the time being keeping it here makes sense. If it turns out to be really massive, we could consider a separate repo, but I think for now, we might find that it's really analogous to notebook and that it fits well in this repo.

I tend to agree with @blink1073 about fewer repos.

**Comment by ellisonbg** [comment]
OK sounds good.

On Wed, Apr 13, 2016 at 11:10 AM, A. Darian notifications@github.com
wrote:

> Since it's going to use so many notebook things, I think for the time
> being keeping it here makes sense. If it turns out to be really massive, we
> could consider a separate repo, but I think for now, we might find that
> it's really analogous to notebook and that it fits well in this repo.
> 
> I tend to agree with @blink1073 https://github.com/blink1073 about
> fewer repos.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/jupyter/jupyter-js-notebook/pull/175#issuecomment-209573213

## 

Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Pull request merged**